hw/ip/i2c/doc/_index.md

```diff
@@ -255,31 +252,26 @@ Other features may also be required for complete SMBus functionality.)
-After this, {{< regref STRETCH_CTRL.STOP_TX >}} or {{< regref STRETCH_CTRL.STOP_ACQ >}}, respectively, is reset so the target device will stretch the clock after acquiring the address of the next transaction intended for it.
+There are 3 conditions in which this design stretches the clock.
 
-{{<wavejson>}}
-{signal: [
-  {name: 'clock', wave: 'p...................'},
-  {name: 'stretch_en_addr', wave: '1...................'},
-  {name: 'stretch_stop', wave: '0.............1.....'},
-  {name: 'bit_ack', wave: '0........1..........'},
-  {name: 'address_match', wave: '0........1..........'},
-  {name: 'SCL host driver', wave: '0z.0..z.0..z.....0..'},
-  {name: 'SCL target driver', wave: 'z.........0....z....'},
-  {name: 'SCL bus', wave: '0u.0..u.0......u.0..'},
-],
- head: {text: 'Clock stretching after matching an address (cycle 10), normal operation is resumed and address is acknowledged (cycle 15)', tick: 1}
-}
-{{</wavejson>}}
-In this diagram, "stretch_en_addr" stands for {{< regref STRETCH_CTRL.EN_ADDR_TX >}} or {{< regref STRETCH_CTRL.EN_ADDR_ACQ >}} whereas "stretch_stop" stands for, respectively, {{< regref STRETCH_CTRL.STOP_TX >}} or {{< regref STRETCH_CTRL.STOP_ACQ >}}.
+#### Stretching after address read
+    - When a target device receives a start, the address and R/W bit are written into the ACQ FIFO.
+    - If there is no space in the ACQ FIFO to receive such a write, the target stretches the clock after the ACK bit and waits for software to make space.
+    - The `acq_full` interrupt is generated to alert software to such a situation.
+
+#### Stretching during write
+    - Similar to the scenario above, if the host tries to write a data byte into the ACQ FIFO when there is no available space, the clock is also stretched after the ACK bit.
+    - The `acq_full` interrupt is generated to alert software to such a situation.
 
-In all cases described above, a target begins to stretch the clock when SCL is held low by a host.
+#### Stretching during read
+    - When a target device receives a start and read command, it may stretch the clock for either of the following two reasons.
+      - If there is no data available to be sent back (TX FIFO empty case), the target stretches the clock until data is made available by software.
+      - If there is more than 1 entry in the ACQ FIFO.
+        - Having more than 1 entry in the ACQ FIFO suggests there is potentially an unhandled condition (STOP / RESTART) or an unhandled command (START) that requires software intervention before the read can proceed.
+    - The `tx_stretch` interrupt is generated to alert software to such a situation.
+
+In all cases described above, a target begins to stretch the clock after the ACK bit.
```

Comment:
This is going to look like it's under the "Stretching during read" label, isn't it? If so, maybe this text should be adjusted to go above the list, or the labels should become part of the list instead?

Reply:
o yeah good point.